script.js

```diff
@@ -1,12 +1,304 @@
+}
+/*2 creates header*/
+
+/*Creates header function that displays the header element */
+function createHeader(displayResult) {
+  const headerElement = document.getElementsByTagName("header"); // Selects the header element based on the Id that it's have
+  const spanDisplayResult = document.createElement("span"); // Creates the span tag.
+  spanDisplayResult.id = "numberOfMatchingTitle"; // Assigns an id to the span tag.
+  spanDisplayResult.innerText = displayResult; // Assign the The total is: 0 /73 elements to span element.
+  headerElement[0].appendChild(spanDisplayResult); // Appended the span to the header element.
+
+  let theForm = document.createElement("form"); // Created the form element.
+  theForm.id = "search-box"; // Assigned it an id called search-box.
+  headerElement[0].appendChild(theForm); //Appended a form to header element.
+
+  let theInputText = document.createElement("input"); // Created the input Element.
+  theInputText.id = "userInput"; //Assigned an id to called userInput.
+  theInputText.type = "text"; //Set the Type to text.
+  theInputText.placeholder = "Search a film..."; //Assigned the placeHolder text to it.
+  theForm.appendChild(theInputText); //Appended the input to form.
+  theInputText.addEventListener("input", (event) => {
+    //eventListener that calls the search function.
+    search(theInputText.value.trim()); // Trimmed value of user input is passed to the search function.
+  });
+}
+function makePageForShows(showData) {
+  rootElem.textContent = ""; // Set the of the root to blank as my the root content is the parent of my elements
+
+  showData.forEach((e) => {
+    // Loops through the episode list
+    setShowHTML(e); // repeats set episode function to create the structure of the page.
+  });
+}
+function setShowHTML(show) {
+  let theContainer = document.createElement("div"); //Creates an div element assign it assign it to the container element.
+  theContainer.setAttribute("value", "");
+  theContainer.setAttribute("class", "xl-col-4 placeHolder"); // Assign the Class of "xl-col-4" from the grid CSS file.
+  rootElem.appendChild(theContainer); // Appended the div element to the parent element of root.
+
+  let theArticle = document.createElement("article"); // Creates an article element.
+  theArticle.id = "article"; //Assigned the id of article to the article.
+  theArticle.setAttribute("class", "artContainer"); // Assigned the class of artContainer to the Article element.
+  theContainer.appendChild(theArticle); // Appended the Article to the div of our placeholder.
+
+  let theTitleTagName = document.createElement("h3"); //Created the a h3 element to hold the title.
+  theArticle.appendChild(theTitleTagName); // Appended the h3 element to the article Container.
+  theTitleTagName.textContent = `${show.name}`; /*-${titleMaker(
+    element.season,
+    element.number
+  )}`;*/ // Set the the text content of the h3 element the object name property called the title maker maker function
```

Comment:
I didn't get what are these comments!
maybe if you don't use it you can make your code more clear by deleting this.

Reply:
yes, this is good advice - once everything is working it is important to go through your code and remove anything that is not used. This is actually a very fun satisfying part of development!